platforms/ide/ide-plugins/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java

```diff
@@ -71,6 +77,11 @@ public void execute(Gradle gradle) {
                 VersionNumber ideaTargetVersion = findIdeaTargetVersion();
                 final boolean useScalaSdk = ideaTargetVersion == null || IDEA_VERSION_WHEN_SCALA_SDK_WAS_INTRODUCED.compareTo(ideaTargetVersion) <= 0;
                 final Collection<Project> scalaProjects = findProjectsApplyingIdeaAndScalaPlugins();
+
+                if (!scalaProjects.isEmpty() && isolatedProjects) {
+                    throw new GradleException(INCOMPATIBLE_WITH_ISOLATED_PROJECTS_MESSAGE);
```

Comment:
💭 With a callback argument, these 3 lines would be moved to `IdeaPlugin`, there would only be a callback call here `onScalaProjects.execute(scalaProjects)`